sdks/java/io/distributedlog/src/main/java/org/apache/beam/sdk/io/distributedlog/DistributedLogIO.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.distributedlog;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Lists;
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.util.List;
+import org.apache.beam.sdk.coders.ByteArrayCoder;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.io.BoundedSource;
+
+/**
+ * The unbounded/bounded source and the sink for <a href="http://distributedlog.io">DistributedLog</a> streams.
```

Comment:
* In fact this is only a bounded source, right?
* Please phrase the documentation and the API of this class consistently with other connectors - e.g. see JdbcIO, MongoDbIO, etc. - use .read() and .write() transform builders with AutoValue, and do not expose the Source class directly (as the implementation may evolve to not use the Source API, and the users' code should keep compiling while this happens).